### PR TITLE
ffmpeg, add version 6.0

### PR DESCRIPTION
**Pull request by Begasus**
Checked with latest VLC (in our repository), build and playback are fine. (not checked on 32bit yet).

### CODE REVIEW

File: media-video/ffmpeg/ffmpeg6-6.0.recipe [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+		# Hack up base headers to make them think we're on GCC2 ABI
+		mkdir -p include_hacks include_hacks/os include_hacks/config
+		pushd include_hacks
+			rm -f os/BeBuild.h config/HaikuConfig.h
+			cp /system/develop/headers/os/BeBuild.h os
+			cp /system/develop/headers/config/HaikuConfig.h config
+			sed -i 's/__GNUC__ == 2/1/g' os/BeBuild.h config/HaikuConfig.h
+		popd
+
+		# Compile gcc_runtime.c (subset of libgcc that GCC 7 utilizes here)
```
Comment (jmairboeck) [inline media-video/ffmpeg/ffmpeg6-6.0.recipe:254]: This comment should refer to GCC 13 now (or generalize it to "modern GCC" or something).

File: media-video/ffmpeg/ffmpeg6-6.0.recipe [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+	ffmpeg${secondaryArchSuffix}_devel
+	ffmpeg5${secondaryArchSuffix}_devel
+	"
+
+# -- HACK! --
+# ffmpeg ~0.10 has perpetually had & caused problems when compiled with GCC 2.
+# (Later versions do not compile with GCC 2 at all.) So instead, we now compile
+# FFmpeg for x86_gcc2 with a modern GCC.
+#
+# This works because GCC's C ABI has not changed between 2 and 7; only the C++
```
Comment (jmairboeck) [inline media-video/ffmpeg/ffmpeg6-6.0.recipe:146]: The same as above here.

Comment (Begasus) [reply media-video/ffmpeg/ffmpeg6-6.0.recipe:146]: Left overs from previous patchset, will adjust :)

File: media-video/ffmpeg/ffmpeg6-6.0.recipe (hunk 1 above)
Comment (Begasus) [reply media-video/ffmpeg/ffmpeg6-6.0.recipe]: Both adjusted, thanks!

**Pull request merged**